### ISSUE TITLE
(Network) issues with zgw/objects endpoints prevent saving form & show "generic" error

### ISSUE THREAD

**Issue by LaurensBurger** [open-formulieren/open-forms#4709]
### Thema / Theme

Form designer

### Omschrijving / Description

As a form designer i want to know if the registration settings in my form are causing issues when saving a form. 

This came to light with #4695 but now also reported by a user with credentials that expired for their ZGW API catalogi end-point. 

The error for these scenario's is generic and positioned on top of the "form" tab of the form, not indicating it originates from the registration tab. 

“De gegevens zijn ongeldig. Los de problemen hieronder op.”.

As a form builder i would not expect this to be network/service related, since i don't expect there to be any requests being made saving a form - therefore finding the solution with only this message is hard.

**Comment by joeribekker** [comment]
Refinement: We should look into the reason of the "credentials expired" part of Laurens description but then also catch this "network" issue and report a proper error.

**Comment by LaurensBurger** [comment]
"credentials expired" was not OF related, but the ZGW services responded with a 401. The user was not aware of the expiration, and was trying to edit something unrelated to the registration in a form.

**Comment by sergei-maertens** [comment]
For the record - a summary of what was discussed with Vasileios and how we approached "solving" this.

The error message that is displayed is really a last resort error message - something goes wrong in the backend that we can't anticipate, and that error (a 401, 403 or maybe even just a programming mistake that causes the backend to crash) is translated into an HTTP 500 error. It could also be that some infrastructure is down and we get HTTP 502/503 errors back, so really, the gist is "we don't know the exact nature of the problem other than it's broken".

*If* there are specific situations that we should handle, we pick those up - ensuring that particular validation errors are captured and displayed in the right location etc. In all the other situations, we must display a generic error message informing the user that something went wrong and that's all we know.

We updated the text of that error message to:

* display more accurately that something went wrong that we did not expect
* removed the text about validation errors - it used to be "accurate" but now we have proper validation error displaying with the offending fields (if we don't, that's a bug)
* added pointers on how the problem can be resolved: either contact a technical administrator and/or consult the outgoing request logs. A link is now included in that message - the important thing is that the outgoing request logs can be a tool to investigate if you have the appropriate permissions.
